src/components/search/search.js

```diff
@@ -19,9 +20,28 @@ class SearchComponent extends React.Component {
     this.setState({ postcode: event.target.value });
   }
 
-  doPostcodeSearch = props => {
-    this.setState({ doSearch: true });
-  };
+  doPostcodeSearch() {
+    console.log(this.state.postcode);
+    if (this.state.postcode) {
+      let url =
+        "http://localhost:8000/api/venue/?postcode=" + this.state.postcode;
+
+      fetch(url)
+        .then(response => {
+          return response.json();
+        })
+        .then(data => {
+          if (data.detail) {
+            console.log(data.detail);
+            this.setState({ error: data.detail });
+          } else {
+            this.setState({ venues: data.slice(0, 9), doSearch: true });
+          }
+        });
+    } else {
+      this.setState({ error: "Please enter a postcode" });
```

Comment:
Would `Please enter a valid postcode` be appropriate?

Reply:
No because this is shown when the user has entered nothing. I will add a comment to clarify